tokio-trace/src/lib.rs

```diff
@@ -345,3 +347,58 @@ pub mod subscriber;
 mod sealed {
     pub trait Sealed {}
 }
+
+/// An enum representing the available verbosity level filters of the logger.
+#[repr(usize)]
+#[derive(Copy, Clone, Debug, Hash)]
+pub enum LevelFilter {
```

Comment:
For forwards-compatibility reasons, we try to avoid having enum types in public APIs. Can this be a struct wrapping an inner private enum type instead? Refer to how the `Level` type in `tokio-trace-core` is implemented:
https://github.com/tokio-rs/tokio/blob/92d51202ef5e6b99004d77300de1aba0f8835513/tokio-trace/tokio-trace-core/src/metadata.rs#L224-L272

Reply:
@hawkw Thanks for this, if we already have this `Level`, perhaps we can just use `Level` for this purpose?

Only `Off` level is missing, do we actually need it? what do you think?

Reply:
> @hawkw Thanks for this, if we already have this `Level`, perhaps we can just use `Level` for this purpose?
> 
> Only `Off` level is missing, do we actually need it? what do you think?

I think we _will_ want an `OFF` level for the static filter, so that `tokio-trace` can be completely disabled in release mode, etc. So, we'll probably need a separate `LevelFilter` type. 

The implementation can be the same as `tokio-trace-core`s `Level` type, but with the addition of an `OFF` level. We will want to implement the [`PartialOrd`](https://doc.rust-lang.org/std/cmp/trait.PartialOrd.html) trait for comparing `LevelFilter`s with `Level`s, (`impl PartialOrd<Level> for LevelFilter`, with the same ordering as for `Level`s, but with `LevelFilter::OFF` considered less than all `Level`s

> It seems to be private struct though.

The `Level` struct is public:
https://github.com/tokio-rs/tokio/blob/92d51202ef5e6b99004d77300de1aba0f8835513/tokio-trace/tokio-trace-core/src/metadata.rs#L129
while the `LevelInner` enum is private:
https://github.com/tokio-rs/tokio/blob/92d51202ef5e6b99004d77300de1aba0f8835513/tokio-trace/tokio-trace-core/src/metadata.rs#L251

We'll want our `LevelFilter` type to follow the same pattern.